CRM/Contribute/Form/AbstractEditPayment.php

```diff
@@ -368,7 +368,7 @@ public function assignProcessors() {
 
     // this required to show billing block
     // @todo remove this assignment the billing block is now designed to be always included but will not show fieldsets unless those sets of fields are assigned
-    $this->assign_by_ref('paymentProcessor', $processor);
+    $this->assign_by_ref('paymentProcessor', $this->_paymentProcessor);
```

Comment:
`$processor` is the variable that iterates in the foreach loop written above and assigns the last iterated value to the `paymentProcessor` var. `$this->_paymentProcessor` holds the correct value that needs to be assigned.